src/Client/ConnectionEstablisher.cpp

```diff
@@ -57,7 +57,9 @@ void ConnectionEstablisher::run(ConnectionEstablisher::TryResult & result, std::
 
         if (!table_to_check || server_revision < DBMS_MIN_REVISION_WITH_TABLES_STATUS)
         {
-            result.entry->forceConnected(*timeouts);
```

Comment:
Previously stale connections have been reestablished in this branch for `s3Cluster()` table function